examples/jsm/exporters/GLTFExporter.js

```diff
@@ -689,80 +689,68 @@ class GLTFWriter {
+		console.warn( 'THREE.GLTFExporter: Merged metalnessMap and roughnessMap textures.' );
 
-		if ( occlusion || roughness || metalness ) {
+		const metalness = metalnessMap?.image;
+		const roughness = roughnessMap?.image;
 
-			const width = Math.max( occlusion?.width || 0, roughness?.width || 0, metalness?.width || 0 );
-			const height = Math.max( occlusion?.height || 0, roughness?.height || 0, metalness?.height || 0 );
+		const width = Math.max( metalness?.width || 0, roughness?.width || 0 );
+		const height = Math.max( metalness?.height || 0, roughness?.height || 0 );
 
-			const canvas = document.createElement( 'canvas' );
-			canvas.width = width;
-			canvas.height = height;
+		const canvas = document.createElement( 'canvas' );
+		canvas.width = width;
+		canvas.height = height;
 
-			const context = canvas.getContext( '2d' );
-			context.fillStyle = '#ffffff';
-			context.fillRect( 0, 0, width, height );
+		const context = canvas.getContext( '2d' );
+		context.fillStyle = '#00ffff';
+		context.fillRect( 0, 0, width, height );
 
-			const composite = context.getImageData( 0, 0, width, height );
+		const composite = context.getImageData( 0, 0, width, height );
 
-			if ( occlusion ) {
+		if ( metalness ) {
 
-				context.drawImage( occlusion, 0, 0, width, height );
+			context.drawImage( metalness, 0, 0, width, height );
 
-				const data = context.getImageData( 0, 0, width, height ).data;
+			const data = context.getImageData( 0, 0, width, height ).data;
 
-				for ( let i = 0; i < data.length; i += 4 ) {
-
-					composite.data[ i ] = data[ i ];
+			for ( let i = 2; i < data.length; i += 4 ) {
 
-				}
+				composite.data[ i ] = data[ i ];
 
 			}
 
-			if ( roughness ) {
+		}
 
-				context.drawImage( roughness, 0, 0, width, height );
+		if ( roughness ) {
 
-				const data = context.getImageData( 0, 0, width, height ).data;
+			context.drawImage( roughness, 0, 0, width, height );
 
-				for ( let i = 1; i < data.length; i += 4 ) {
+			const data = context.getImageData( 0, 0, width, height ).data;
 
-					composite.data[ i ] = data[ i ];
+			for ( let i = 1; i < data.length; i += 4 ) {
 
-				}
+				composite.data[ i ] = data[ i ];
 
 			}
 
-			if ( metalness ) {
-
-				context.drawImage( metalness, 0, 0, width, height );
-
-				const data = context.getImageData( 0, 0, width, height ).data;
-
-				for ( let i = 2; i < data.length; i += 4 ) {
-
-					composite.data[ i ] = data[ i ];
-
-				}
+		}
 
-			}
+		context.putImageData( composite, 0, 0 );
 
-			context.putImageData( composite, 0, 0 );
+		//
 
-			const texture = new Texture( canvas );
+		const reference = metalnessMap || roughnessMap;
 
-			texture.userData.mimeType = material.roughnessMap.userData.mimeType || material.metalnessMap.userData.mimeType || material.aoMap.userData.mimeType;
+		const texture = reference.clone();
 
-			return texture;
+		// TODO Use new Source() instead?
+		texture.source = new Texture( canvas ).source;
```

Comment:
@Mugen87

I was looking for a nice way of copying all the properties of a texture and change the source but I ended up doing this hack because seems like we're currently not exposing `THREE.Source`.

What do you think?

Reply:
Initially, `THREE.Source` was meant to be an internal thing.

However, looking at your code it would be indeed nicer to expose `THREE.Source` and use it in the mentioned line.